zerver/lib/upload.py

```diff
@@ -470,13 +471,15 @@ def copy_avatar(self, source_profile: UserProfile, target_profile: UserProfile)
     def get_avatar_url(self, hash_key: str, medium: bool=False) -> str:
         bucket = settings.S3_AVATAR_BUCKET
         medium_suffix = "-medium.png" if medium else ""
+        url = urllib.parse.urlparse(self.avatar_bucket.meta.client.meta.endpoint_url)
```

Comment:
I think this would be cleaner written if we moved the whole `f"https://{bucket}.{url.netloc}/` computation to be `self.avatar_bucket_url`; we could delete the `bucket` local variable here, etc.  

And potentially also define `self.uploads_bucket_url`.

Reply:
Agreed. I threw this together rather quickly last night, just to get your thoughts on the approach. 

I’ll clean things up.

Reply:
Agreed. I threw this together rather quickly last night, just to get your thoughts on the approach. 

I’ll clean things up.